### PR TITLE
add 'flavor' field to addon schema

### LINKED ISSUE #1098: update addon schema to allow "flavor" selection
Hello,

I am try to deploy autoscaler addon in HA mode, but the provider does not support the "flavor" bloc. Could it be possible to update the addon schema to add this bloc, or should i add it and make a pull request ?

### PR DESCRIPTION
**Pull request by Ya-S**
This pull request adds "flavor" field to the addon schema to allow selection of the flavor (single node/HA)

### CODE REVIEW

File: huaweicloud/resource_huaweicloud_cce_addon_v3.go [hunk 1]
```diff
@@ -71,14 +71,19 @@ func ResourceCCEAddonV3() *schema.Resource {
...
+						},
 					},
 				},
 			},
 		},
 	}
 }
 
-func getValuesValues(d *schema.ResourceData) (basic, custom map[string]interface{}, err error) {
+func getValuesValues(d *schema.ResourceData) (basic, custom map[string]interface{}, flavor map[string]interface{}, err error) {
```
Comment (Jason-Zhang9309) [inline huaweicloud/resource_huaweicloud_cce_addon_v3.go]: use `basic, custom, flavor map[string]interface{}` could be better

**Comment by Jason-Zhang9309** [comment]
Hi,
Thanks for your efforts. 
I have some suggestions for you. 

 As "github.com/huaweicloud/golangsdk/openstack/cce/v3/addons" is a imported package, you have to modify the golangsdk first(https://github.com/huaweicloud/golangsdk).

After the pull request in golangsdk is merged, use commands:
`go get github.com/huaweicloud/golangsdk`
`go mod tidy`
`go mod vendor` 
to update the go.mod, go.sum and vendor files of you local repo.

Then update you pull request in this repo.

**Comment by ShiChangkuo** [comment]
@Ya-S Please **rebase** your respository to make the CI successful.

Also, you should following the PULL_REQUEST_TEMPLATE in the description of your pull request.

The Acceptance Steps is recommended before merging it. Did you test it well in Acceptance or your local tf script?

File: huaweicloud/resource_huaweicloud_cce_addon_v3.go [hunk 2]
```diff
@@ -71,14 +71,19 @@ func ResourceCCEAddonV3() *schema.Resource {
 							Optional: true,
 							Elem:     &schema.Schema{Type: schema.TypeString},
 						},
+						"flavor": {
```
Comment (ShiChangkuo) [inline huaweicloud/resource_huaweicloud_cce_addon_v3.go:79]: the docs should be updated as while.

**Comment by Lance52259** [comment]
This PR shouldn't contain golang sdk modifications that are not related to the current resource.
you can use `git rm xxx` to remove them or use `git checkout -- xxx` when you finish `go mod vendor`.

**Comment by ShiChangkuo** [comment]
thanks for you effort!

**Pull request merged**